### PR TITLE
feat(connectivity): Check rds cfg domains for pod-to-url

### PR DESCRIPTION
**Pull request by johnsonshi**
Depends on functionality implemented in #64.

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

### CODE REVIEW

**Comment by johnsonshi** [comment]
Depends on #64.

**Review by SanyaKochhar — APPROVED** [review]

**Review by draychev — CHANGES_REQUESTED** [review]
I see that this PR is for `pod-to-url`, but the changes are to the `pkg/connectivity/pod-to-pod.go` file.

It seems to me that the changes should be in `pkg/connectivity/pod-to-url.go` around here:
https://github.com/openservicemesh/osm-health/blob/dde34257d45b321508553186b25ed4409798dc87/pkg/connectivity/pod-to-url.go#L14


The way I was thinking about this - if you did some pod to `http://bookstore.bookstore/buy-a-book`:
  1. we'll resolve a bunch of pods for `bookstore.bookstore` and run the existing `pod-to-pod` for each pod discovered
  2. we'll check routes for each pod in #1 for the route provided via the CLI - `/buy-new-book`

File: pkg/connectivity/pod-to-pod.go [hunk 1]
```diff
@@ -79,10 +79,10 @@ func PodToPod(fromPod *corev1.Pod, toPod *corev1.Pod, osmControlPlaneNamespace s
 		envoy.HasSpecificEndpoint(srcConfigGetter, toPod),
 
 		// Check whether the source Pod has an outbound dynamic route config domain that matches the destination Pod.
-		envoy.HasOutboundDynamicRouteConfigDomainCheck(srcConfigGetter, toPod),
+		envoy.HasOutboundDynamicRouteConfigDomainPodCheck(srcConfigGetter, toPod),
 
 		// Check whether the destination Pod has an inbound dynamic route config domain that matches the source Pod.
-		envoy.HasInboundDynamicRouteConfigDomainCheck(dstConfigGetter, fromPod),
+		envoy.HasInboundDynamicRouteConfigDomainPodCheck(dstConfigGetter, fromPod),
```
Comment (johnsonshi) [inline pkg/connectivity/pod-to-pod.go:85]: @draychev - This is just renaming the method names - the bulk of this PR lies in pod-to-url.

**Review by draychev — APPROVED** [review]

**Comment by draychev** [comment]
Happy to take a different approach - perhaps let's make this work, then dedupe and reorganize the code base.

**Pull request merged**